zrml/prediction-markets/src/migrations.rs

```diff
@@ -1 +1,190 @@
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            if market.status == MarketStatus::Resolved {
+                Disputes::<T>::remove(market_id);
+                total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            }
+        }
+        StorageVersion::new(PREDICTION_MARKETS_NEXT_STORAGE_VERSION).put::<Pallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+
+        log::info!("Completed storage removal of RemoveDisputesOfResolvedMarkets");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        for (market_id, market) in T::MarketCommons::market_iter() {
+            if market.status == MarketStatus::Resolved {
+                let disputes = Disputes::<T>::get(market_id);
+                assert_eq!(disputes.len(), 0);
+            }
+        }
+
+        let prediction_markets_storage_version = StorageVersion::get::<Pallet<T>>();
+        assert_eq!(
+            prediction_markets_storage_version, PREDICTION_MARKETS_NEXT_STORAGE_VERSION,
+            "found unexpected prediction-markets pallet storage version. Found: {:?}. Expected: \
+             {:?}",
+            prediction_markets_storage_version, PREDICTION_MARKETS_NEXT_STORAGE_VERSION,
+        );
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{mock::*, MomentOf};
+    use frame_support::assert_ok;
+    use orml_traits::MultiCurrency;
+    use zeitgeist_primitives::{
+        constants::{BASE, MILLISECS_PER_BLOCK},
+        types::{
+            Asset, BlockNumber, MarketCreation, MarketDispute, MarketDisputeMechanism,
+            MarketPeriod, MarketType, MultiHash, OutcomeReport, ScoringRule,
+        },
+    };
+
+    #[test]
+    fn test_on_runtime_upgrade_on_untouched_chain() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            RemoveDisputesOfResolvedMarkets::<Runtime>::on_runtime_upgrade();
+        });
+    }
+
+    #[test]
+    fn on_runtime_upgrade_updates_storage_versions() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            RemoveDisputesOfResolvedMarkets::<Runtime>::on_runtime_upgrade();
+            assert_eq!(
+                StorageVersion::get::<Pallet<Runtime>>(),
+                PREDICTION_MARKETS_NEXT_STORAGE_VERSION
+            );
+        });
+    }
+
+    #[test]
+    fn test_on_runtime_upgrade_with_sample_markets() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            let _ = Currency::deposit(Asset::Ztg, &ALICE, 1_000 * BASE);
+
+            // Markets which end here will have to be closed on migration:
```

Comment:
[nitpick] You don't really depend on how long these markets run since you set them to `Resolved` anyways.

```suggestion
```